libs/domain/user/address-list/src/address-list.component.ts

```diff
@@ -28,85 +29,35 @@ export class AddressListComponent extends ContentMixin<AddressListItemOptions>(
 
   protected addressService = resolve(AddressService);
 
-  protected selectedAddress$ = new ReplaySubject<Address | null>(1);
-
   @asyncState()
-  protected selectedAddress = valueType(this.selectedAddress$);
+  protected addresses = valueType(this.addressService.getAddresses());
+
+  @state()
+  protected selectedAddressId?: string;
 
   @subscribe()
-  protected handleSelectedAddress$ = this.selectedAddress$.pipe(
-    tap((address) => {
-      if (address) {
-        this.dispatchEvent(
-          new CustomEvent(SELECT_EVENT, {
-            bubbles: true,
-            composed: true,
-            detail: { address },
-          })
+  protected addressesSubscription = this.addressService.getAddresses().pipe(
+    tap((addresses) => {
+      if (!this.componentOptions?.selectable) return;
```

Comment:
Much better, but still. This subscription will trigger only when addresses change, the same situation if for some reason addresses will be fetched earlier then options, or if options change after, it has no effect. Unfortunately need to combine them to keep stable behaviour